### PR TITLE
[Enhancement] support limit offset,limit

### PR DESCRIPTION
**Pull request by Seaven**
Fixes #issue
Support `limit offset, limit; ` SQL, just promise right rows, not stable result. Transform Exchange to Merge-Exchange when has offset, and sort by `constant expr(1)`, like:
```
select * from xx limit 10, 20 ->  select * from xx order by '1'(constant expr) limit 10, 20;
```

```
MySQL td> explain select COUNT(*) from (select * from t0 limit 10, 20) xx;

+-----------------------------------------+
| Explain String                          |
+-----------------------------------------+
| PLAN FRAGMENT 0                         |
|  OUTPUT EXPRS:4: count                  |
|   PARTITION: UNPARTITIONED              |
|  ......                                 |
|   2:MERGING-EXCHANGE                    |
|      offset: 10                         |
|      limit: 20                          |
|                                         |
| PLAN FRAGMENT 1                         |
|  OUTPUT EXPRS:                          |
|   PARTITION: RANDOM                     |
|  ......                                 |
|
|   1:Project                             |
|   |  <slot 6> : 1                       |
|   |  limit: 30                          |
|   |                                     |
|   0:OlapScanNode                        |
|      TABLE: t0                          |
|  ......                                 |
|      limit: 30                          |
+-----------------------------------------+
```

* Remove `Limit` from GatherProperty, `limit` shouldn't be guaranteed by property
* Add `offset` and `limit` handle process in `PlanFragment.visitPhysicalLimit`
* Add `ConstChunkMerger` in BE, to handle merge chunk by constant expression

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 44 / 45 (97.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithLimitRule.java|6|7|85.71%|[90]|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownLimitCTEAnchor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|26|26|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalLimitOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownProjectLimitRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalLimitOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SplitLimitRule.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 44 / 45 (97.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithLimitRule.java|6|7|85.71%|[90]|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownLimitCTEAnchor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|26|26|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalLimitOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownProjectLimitRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalLimitOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SplitLimitRule.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 58 / 60 (96.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|1|2|50.00%|[369]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|24|25|96.00%|[1586]|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownLimitCTEAnchor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneGroupByKeysRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalLimitOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithLimitRule.java|21|21|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownProjectLimitRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalLimitOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SplitLimitRule.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 58 / 60 (96.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/EnforceAndCostTask.java|1|2|50.00%|[369]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|24|25|96.00%|[1586]|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownLimitCTEAnchor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneGroupByKeysRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalLimitOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeLimitWithLimitRule.java|21|21|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownProjectLimitRule.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalLimitOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SplitLimitRule.java|2|2|100.00%|[]|

**Comment by Seaven** [comment]
@kangkaisen Ignore ADMIN TEST, it's an error case

File: be/src/runtime/sorted_chunks_merger.cpp [hunk 1]
```diff
@@ -347,4 +347,51 @@ Status CascadeChunkMerger::get_next(ChunkUniquePtr* output, std::atomic<bool>* e
...
+Status ConstChunkMerger::init(const std::vector<ChunkProvider>& providers, const std::vector<ExprContext*>* sort_exprs,
+                              const std::vector<bool>* sort_orders, const std::vector<bool>* null_firsts) {
+    for (const auto& expr : *sort_exprs) {
+        DCHECK(expr->root()->is_constant());
+    }
+    _providers = providers;
+    return Status::OK();
+}
+
+bool ConstChunkMerger::is_data_ready() {
```
Comment (fzhedu) [inline be/src/runtime/sorted_chunks_merger.cpp:369]: const

Comment (Seaven) [reply be/src/runtime/sorted_chunks_merger.cpp:369]: Just keep same interface, I'm not sure other impelemention is const

**Review by stdpain — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Pull request merged**